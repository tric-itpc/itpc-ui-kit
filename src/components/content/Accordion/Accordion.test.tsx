import React from "react"

import { fireEvent, render, screen } from "@testing-library/react"

import {
  Accordion,
  AccordionArrow,
  AccordionBody,
  AccordionHeader,
  AccordionItem,
} from "./index"

describe("Компоненты Accordion", () => {
  test("Рендер Accordion с дефолтными пропсами (снимок)", () => {
    const { container } = render(
      <Accordion>
        <AccordionItem>Item 1</AccordionItem>
      </Accordion>
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  test("Рендер AccordionItem с контентом", () => {
    render(<AccordionItem>Контент</AccordionItem>)
    expect(screen.getByText("Контент")).toBeInTheDocument()
  })

  test("Рендер AccordionHeader с кликабельным состоянием", () => {
    const onPressMock = jest.fn()
    render(<AccordionHeader onPress={onPressMock}>Заголовок</AccordionHeader>)

    const header = screen.getByText("Заголовок")
    fireEvent.click(header)

    expect(onPressMock).toHaveBeenCalledTimes(1)
  })

  test("Рендер AccordionArrow с правильной ориентацией в зависимости от isOpened", () => {
    const { container, rerender } = render(<AccordionArrow isOpened />)

    const iconArrow = container.querySelector(".itpc-accordion-arrow")

    expect(iconArrow?.querySelector("i")).toHaveClass(
      "itpc-icon__arrow_orientation_top"
    )

    rerender(<AccordionArrow isOpened={false} />)

    expect(iconArrow?.querySelector("i")).toHaveClass(
      "itpc-icon__arrow_orientation_bottom"
    )
  })

  test("Рендер AccordionBody с правильным классом в зависимости от isOpened", () => {
    const { container, rerender } = render(
      <AccordionBody isOpened>Контент</AccordionBody>
    )

    const accordionBody = container.querySelector(".itpc-accordion-body")

    expect(accordionBody).toHaveClass("itpc-accordion-body_opened")

    rerender(<AccordionBody isOpened={false}>Контент</AccordionBody>)

    expect(accordionBody).not.toHaveClass("itpc-accordion-body_opened")
  })

  test("Рендер AccordionHeader с активным состоянием (isActive)", () => {
    render(<AccordionHeader isActive>Активный заголовок</AccordionHeader>)
    expect(screen.getByText("Активный заголовок")).toHaveClass(
      "itpc-accordion-header_active"
    )
  })

  test("Рендер AccordionHeader с состоянием открытия (isOpened)", () => {
    render(<AccordionHeader isOpened>Заголовок открыт</AccordionHeader>)
    expect(screen.getByText("Заголовок открыт")).toHaveClass(
      "itpc-accordion-header_opened"
    )
  })

  test("AccordionBody скрыт при isOpened=false", () => {
    const { container, rerender } = render(
      <AccordionBody isOpened={false}>Контент</AccordionBody>
    )

    const accordionBody = container.querySelector(".itpc-accordion-body")

    expect(accordionBody).not.toHaveClass("itpc-accordion-body_opened")

    rerender(<AccordionBody isOpened>Контент</AccordionBody>)

    expect(accordionBody).toHaveClass("itpc-accordion-body_opened")
    expect(accordionBody).not.toHaveStyle("display: none")
  })

  test("AccordionBody показан при isOpened=true", () => {
    render(<AccordionBody isOpened>Контент</AccordionBody>)
    const body = screen.getByText("Контент")
    expect(body).toBeVisible()
  })

  test("Accordion передает кастомный класс", () => {
    render(
      <Accordion className="custom-class">
        <AccordionItem>Item 1</AccordionItem>
      </Accordion>
    )
    expect(screen.getByText("Item 1").parentElement).toHaveClass("custom-class")
  })

  test("AccordionArrow снимок", () => {
    const { container } = render(<AccordionArrow isOpened={false} />)
    expect(container.firstChild).toMatchSnapshot()
  })

  test("AccordionBody снимок с isOpened=true", () => {
    const { container } = render(
      <AccordionBody isOpened>Контент</AccordionBody>
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  test("AccordionHeader снимок с isActive и isOpened", () => {
    const { container } = render(
      <AccordionHeader isActive isOpened>
        Заголовок с состоянием
      </AccordionHeader>
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  test("AccordionItem снимок с контентом", () => {
    const { container } = render(<AccordionItem>Item 1</AccordionItem>)
    expect(container.firstChild).toMatchSnapshot()
  })
})
