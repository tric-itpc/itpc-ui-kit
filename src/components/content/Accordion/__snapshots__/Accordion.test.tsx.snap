// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компоненты Accordion AccordionArrow снимок 1`] = `
<span
  class="itpc-accordion-arrow"
>
  <i
    class="itpc-icon__arrow itpc-icon__arrow_orientation_bottom"
  >
    <svg
      height="8"
      viewBox="0 0 12 8"
      width="12"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M10.5134 0L6 4.82292L1.48659 0L0 1.58854L6 8L12 1.58854L10.5134 0Z"
      />
    </svg>
  </i>
</span>
`;

exports[`Компоненты Accordion AccordionBody снимок с isOpened=true 1`] = `
<div
  class="itpc-accordion-body itpc-accordion-body_opened"
>
  Контент
</div>
`;

exports[`Компоненты Accordion AccordionHeader снимок с isActive и isOpened 1`] = `
<div
  class="itpc-accordion-header itpc-accordion-header_opened itpc-accordion-header_active"
  id="itpc-accordion-header"
>
  Заголовок с состоянием
</div>
`;

exports[`Компоненты Accordion AccordionItem снимок с контентом 1`] = `
<div
  class="itpc-accordion-item"
>
  Item 1
</div>
`;

exports[`Компоненты Accordion Рендер Accordion с дефолтными пропсами (снимок) 1`] = `
<div
  class="itpc-accordion"
>
  <div
    class="itpc-accordion-item"
  >
    Item 1
  </div>
</div>
`;
